taxcalc/records.py

```diff
@@ -290,253 +317,223 @@ def set_current_year(self, new_current_year):
-        times_equal(self.e03270, self.BF.ACPIM[year])
-        times_equal(self.e03240, self.BF.AGDPN[year])
-        times_equal(self.e03290, self.BF.ACPIM[year])
-        times_equal(self.e03300, self.BF.ATXPY[year])
-        times_equal(self.e03400, self.BF.ATXPY[year])
-        times_equal(self.e03500, self.BF.ATXPY[year])
-        # Adjusted Gross Income is a calculated field
-        times_equal(self.e00100, 1.)
-        times_equal(self.p04470, 1.)
-        times_equal(self.e04250, 1.)
-        times_equal(self.e04600, 1.)
-        times_equal(self.e04800, 1.)
-        times_equal(self.e05100, 1.)
-        times_equal(self.e05200, 1.)
-        times_equal(self.e05800, 1.)
-        times_equal(self.e06000, 1.)
-        times_equal(self.e06200, 1.)
-        times_equal(self.e06300, 1.)
-        times_equal(self.e09600, 1.)
-        times_equal(self.e07180, 1.)
-        times_equal(self.e07200, 1.)
-        times_equal(self.e07220, 1.)
-        times_equal(self.e07230, self.BF.ATXPY[year])
-        times_equal(self.e07240, self.BF.ATXPY[year])
-        times_equal(self.e07260, self.BF.ATXPY[year])
-        times_equal(self.e07300, self.BF.ABOOK[year])
-        times_equal(self.e07400, self.BF.ABOOK[year])
-        times_equal(self.e07600, 1.)
-        times_equal(self.p08000, self.BF.ATXPY[year])
-        times_equal(self.e07150, 1.)
-        times_equal(self.e06500, 1.)
-        times_equal(self.e08800, 1.)
-        times_equal(self.e09400, 1.)
-        times_equal(self.e09700, self.BF.ATXPY[year])
-        times_equal(self.e09800, self.BF.ATXPY[year])
-        times_equal(self.e09900, self.BF.ATXPY[year])
-        times_equal(self.e10300, 1.)
-        times_equal(self.e10700, self.BF.ATXPY[year])
-        times_equal(self.e10900, self.BF.ATXPY[year])
-        times_equal(self.e59560, self.BF.ATXPY[year])
-        times_equal(self.e59680, self.BF.ATXPY[year])
-        times_equal(self.e59700, self.BF.ATXPY[year])
-        times_equal(self.e59720, self.BF.ATXPY[year])
-        times_equal(self.e11550, self.BF.ATXPY[year])
-        times_equal(self.e11070, self.BF.ATXPY[year])
-        times_equal(self.e11100, self.BF.ATXPY[year])
-        times_equal(self.e11200, self.BF.ATXPY[year])
-        times_equal(self.e11300, self.BF.ATXPY[year])
-        times_equal(self.e11400, self.BF.ATXPY[year])
-        times_equal(self.e11570, self.BF.ATXPY[year])
-        times_equal(self.e11580, self.BF.ATXPY[year])
-        times_equal(self.e11581, self.BF.ATXPY[year])
-        times_equal(self.e11582, self.BF.ATXPY[year])
-        times_equal(self.e11583, self.BF.ATXPY[year])
-        times_equal(self.e10605, self.BF.ATXPY[year])
-        times_equal(self.e11900, 1.)
-        times_equal(self.e12000, 1.)
-        times_equal(self.e12200, 1.)
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-locals
+        # local copies of blowup factors for specified year:
+        AWAGE = self.BF.AWAGE[year]
```

Comment:
I'm surprised by a few things in this section. Here are a few general comments:

- the comment on line 325 is not correct. The variable assignments on lines 325-341 don't make copies of the values on the right hand side. Therefore, I find this section of code quite verbose, since I don't see what is gained by these variable assignments.

- my first comment is just a style issue (aside from the issue with the comment on line 325), and it seems reasonable that if someone would prefer to use `ADIVS` rather than `self.BF.ADIVS[year]`,  as long as he or she is willing to type out all of those variables, there is obviously no difference in how the code executes statements with either `ADIVS` or `self.BF.ADVIS[year]`. This comment is on the more serious performance implications of the changes here. The `time_equal` function used `np.multiply` to do in-place modification of the numpy arrays. Prior to using this `times_equal` function, all of the updates to variables here used the `*=` operator to do similar in-place operations. That is, they were all of the form:

```
    self.e00200 *= self.BF.AWAGE[year]
```

rather than 
```
  self.e00200  = self.e00200 * self.BF.AWAGE[year]
```
The reason to prefer the former is that it does not allocate new memory to hold the results of the multiplication and then assign this new object to the attribute `e00200` on the `self` object, whereas the latter choice does exactly that. So, in the code below, every assignment would involve a new memory allocation, which would then be filled with the results of the multiplication, and this object would be assigned to the attribute on the left, and the previous numpy array would no longer be referred to by anything, and so it would have to be garbage collected at some point.

- similarly, the `np.where` assignments previously used a similar technique to avoid new memory allocation. The old code did as follows:

```
  self.e00900p[:] = np.where(self.e00900p >= 0,
                                  self.e00900p *
                                  self.BF.ASCHCI[year],
                                  self.e00900p *
                                  self.BF.ASCHCL[year])
```

The left-hand side uses the `[:]` indexing style to essentially say "assign into all of the indices here", so that the values created on the right-hand side are place there. Without the `[:]`, a new numpy object is created on the right hand side, this is assigned to the attribute `e00900` on the object `self`, and then the previous numpy array referred to is no longer referred to by anyone, and so it must be garbage collected. So it is more efficient to use the `[:]` syntax on these assignments.